Civi/Api4/Generic/Traits/ArrayQueryActionTrait.php

```diff
@@ -166,6 +166,8 @@ public static function filterCompare(array $row, array $condition, int $index =
 
       case 'REGEXP':
       case 'NOT REGEXP':
+      case 'REGEXP BINARY':
+      case 'NOT REGEXP BINARY':
         $pattern = '/' . str_replace('/', '\\/', $expected) . '/';
         return !preg_match($pattern, $value) == ($operator != 'REGEXP');
```

Comment:
To accommodate both variants you'd need to change this to 
```suggestion
        return !preg_match($pattern, $value) == str_starts_with($operator, 'NOT');
```